torch/ao/quantization/quantize.py

```diff
@@ -435,6 +435,7 @@ def prepare_qat(model, mapping=None, inplace=False, allow_list=None):
         allow_list: a set that lists out allowable modules to be propagated with qconfig
     """
     torch._C._log_api_usage_once("quantization_api.quantize.prepare_qat")
+    assert model.training, "prepare_qat only works on models in training mode"
```

Comment:
Does it make sense to not have this restriction on the model? our APIs should work on the model irrespective of it being in `train`/`eval` mode.

Reply:
we do have this assumption currently, although it's implicit. I think we can relax the constraint later

Reply:
we will need to refactor these to remove the constraint: https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fuser_method_mappings.py#L32, will add the PR in this stack